lib/bgs/dependents.rb

```diff
@@ -0,0 +1,229 @@
+        # I think we need the death_location instead of creating an address
+        # There is no support in the API for death location
+        # create_address(@proc_id, participant[:vnp_ptcpnt_id], death_info['location'])
+
+        @dependents << serialize_dependent_result(
+          participant,
+          relationship_types[:participant],
+          relationship_types[:family],
+          { type: 'death' }
+        )
+      end
+    end
+
+    def report_divorce
+      divorce_info = format_divorce_info
+      participant = create_participant(@proc_id)
+      create_person(@proc_id, participant[:vnp_ptcpnt_id], divorce_info)
+
+      @dependents << serialize_dependent_result(
+        participant,
+        'Spouse',
+        'Spouse',
+        {
+          divorce_state: divorce_info['divorce_state'],
+          divorce_city: divorce_info['divorce_city'],
+          marriage_termination_type_code: divorce_info['marriage_termination_type_code']
+        }
+      )
+    end
+
+    def report_stepchild
+      @dependents_application['step_children'].each do |stepchild_info|
+        step_child_formatted = format_stepchild_info(stepchild_info)
+        participant = create_participant(@proc_id)
+        create_person(@proc_id, participant[:vnp_ptcpnt_id], step_child_formatted)
+        generate_address(participant[:vnp_ptcpnt_id], stepchild_info['address'])
+
+        @dependents << serialize_dependent_result(
+          participant,
+          'Child',
+          'Stepchild',
+          {
+            living_expenses_paid: step_child_formatted['living_expenses_paid'],
+            'type': 'stepchild'
+          }
+        )
+      end
+    end
+
+    def report_child_event(event_type)
+      formatted_child_info = format_child_event(event_type)
+      participant = create_participant(@proc_id)
+
+      create_person(@proc_id, participant[:vnp_ptcpnt_id], formatted_child_info)
+
+      @dependents << serialize_dependent_result(
+        participant,
+        'Child',
+        'Other',
+        {
+          'event_date': formatted_child_info['event_date'],
+          'type': event_type
+        }
+      )
+    end
+
+    def format_child_event(event_type)
+      return format_child_marriage_info(@dependents_application['child_marriage']) if event_type == 'child_marriage'
+
+      format_child_not_attending(@dependents_application['child_stopped_attending_school'])
+    end
+
+    def report_674
+      formatted_674_info = format_674_info
+      student_address = @dependents_application['student_address_marriage_tuition']['address']
+      participant = create_participant(@proc_id)
+      create_person(@proc_id, participant[:vnp_ptcpnt_id], formatted_674_info)
+      generate_address(participant[:vnp_ptcpnt_id], student_address)
+
+      @dependents << serialize_dependent_result(
+        participant,
+        'Child',
+        'Other',
+        { 'type': '674' }
+      )
+    end
+
+    def format_674_info
```

Comment:
Could these formatting methods be pulled out? It may clean up this service class a bit

Reply:
Sure thing. Did you have an implementation in mind? Maybe a helper/formatter module?

Reply:
Great suggestion. I like this much better, thank you.